sommelier => 20220207

supechicken opened the pull request:
## Tested architectures
- [x] x86_64

## Changes
- All changes from #6405
- Move all scripts to https://github.com/chromebrew/crew-package-sommelier (more code comments will be added there)
- The `sommelier` source becomes a git submodule of the new repo now
- `sommelierd`: moved to ruby for better code management
- `sommelierd`: New command: `--log` (see https://github.com/chromebrew/crew-package-sommelier/blob/main/sommelierd)

## Working
- [x] Wayland acceleration
![Screenshot 2022-03-16 12 44 17 AM](https://user-images.githubusercontent.com/73414985/158442251-7b3123a8-e2aa-4500-ae29-99a55f8aa2ea.png)
- [x] X11 acceleration
```
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) UHD Graphics (CML GT2) (0x9b41)
    Version: 21.3.7
    Accelerated: yes
    Video memory: 3072MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) UHD Graphics (CML GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 21.3.7 (git-d5ec846bc8)
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 21.3.7 (git-d5ec846bc8)
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile
```

## Run the following to test this pull request
```shell
CREW_TESTING_REPO=https://github.com/supechicken/chromebrew.git \
CREW_TESTING_BRANCH=update_sommelier \
CREW_TESTING=1 \
crew update
crew reinstall sommelier
```

uberhacker commented on packages/sommelier.rb:
```diff
@@ -3,7 +3,7 @@
   description 'Sommelier works by redirecting X11 programs to the built-in ChromeOS Exo Wayland server.'
   homepage 'https://chromium.googlesource.com/chromiumos/platform2/+/HEAD/vm_tools/sommelier/'
-  version '20211129'
+  version '20220207'
```
Should the `git_branch` match the version?

supechicken replied:
Fixed

supechicken commented on packages/sommelier.rb:
```diff
@@ -3,7 +3,7 @@
+  version '20220207'
   license 'BSD-Google'
   compatibility 'all'
   source_url 'https://github.com/supechicken/crew-sommelier.git'
```
Oh, I forgot to mention this in `Changes` part, I moved all scripts (wrapper script, env script, `sommelierrc` and `sommelierd`) to a separate repo (https://github.com/chromebrew/crew-package-sommelier) for improving code readability (it is hard to see scripts in heredoc without syntax highlighting).

Also the sommelier source is now being one of the git submodules of the new repo (`/platform2`) (another submodule is `https://github.com/supechicken/sommelier-chromiumos-patch`, which contain all necessary patches for running sommelier on CrOS dev shell)

satmandu commented:
I would suggest creating a patches folder in https://github.com/chromebrew/crew-package-sommelier and put all the patches inside there without the patches pointing to a separate submodule repo. Then create a tagged release of `crew-package-sommelier`. That way you can have `git_hashtag version` in the sommelier package file and we can just iterate versions as new tags which tie all connected files and patches together.

I'm ok with the setup at https://github.com/chromebrew/crew-package-sommelier otherwise.

satmandu reviewed (changes requested):
See comments.

satmandu commented:
Yay for acceleration still working!

supechicken commented:
### Changes in [6089391](https://github.com/skycocker/chromebrew/pull/6857/commits/608939109ca2b3d5f90c79f6ab3bf17babca027f)
- Switch to version tag (`20220207`)
- Add code comments (chromebrew/crew-package-sommelier)
- Move `.Xauthority` file to `/home/chronos/.Xauthority`, which is the default location on Chrome OS (see `/etc/profile.d/xauthority.sh `)

satmandu commented:
It isn't working for me on `x86_64`.

satmandu commented:
`/tmp/sommelier.log`
```
[somm_wl]: process running with PID 10285
[somm_x11]: process running with PID 10284
[somm_x11]: warning: virtwl-dmabuf driver not supported by host, using virtwl instead
[somm_x11]: error: Kernel is too old for Iris. Consider upgrading to kernel v4.16.
[somm_x11]:
[somm_x11]: _XSERVTransmkdir: ERROR: euid != 0,directory /tmp/.X11-unix will not be created.
[somm_x11]: error: Kernel is too old for Iris. Consider upgrading to kernel v4.16.
[somm_x11]:
[somm_x11]: Refusing to try glamor on llvmpipe
[somm_x11]: EGL setup failed, disabling glamor
[somm_x11]: Failed to initialize glamor, falling back to sw
[somm_x11]: The XKEYBOARD keymap compiler (xkbcomp) reports:
[somm_x11]: > Warning:          Unsupported maximum keycode 569, clipping.
[somm_x11]: >                   X11 cannot support keycodes above 255.
[somm_x11]: > Error:            Key <CAPS> added to map for multiple modifiers
[somm_x11]: >                   Using Mod3, ignoring Lock.
[somm_x11]: Errors from xkbcomp are not fatal to the X server
[somm_x11]: localuser:root being added to access control list
```

supechicken commented:
Does the `$MESA_LOADER_DRIVER_OVERRIDE` variable set correctly in your system?

satmandu commented:
I didn't have `source /usr/local/etc/profile` in ~/.bashrc.  reloading crosh fixed it!

satmandu commented:
FYI acceleration isn't working for me though...

satmandu commented:
Acceleration started working after a reboot. :)

supechicken commented:
> I would suggest creating a patches folder in https://github.com/chromebrew/crew-package-sommelier . Then put all the patches inside there without the patches folder itself pointing to a separate submodule repo. Then create a tagged release of `crew-package-sommelier`. That way you can have `git_hashtag version` in the sommelier package file and we can just iterate versions as new tags which tie all connected files and patches together.
> 
> I'm ok with the setup at https://github.com/chromebrew/crew-package-sommelier otherwise.

Updated and moved patches into the repo (I forgot to say it earlier😅) (see chromebrew/crew-package-sommelier#4a40bc2)

satmandu commented:
> > I would suggest creating a patches folder in https://github.com/chromebrew/crew-package-sommelier . Then put all the patches inside there without the patches folder itself pointing to a separate submodule repo. Then create a tagged release of `crew-package-sommelier`. That way you can have `git_hashtag version` in the sommelier package file and we can just iterate versions as new tags which tie all connected files and patches together.
> > I'm ok with the setup at https://github.com/chromebrew/crew-package-sommelier otherwise.
> 
> Updated and moved patches into `crew-package-sommelier` (I forgot to say it earliersweat_smile) (see chromebrew/crew-package-sommelier#4a40bc2)

I noticed. :)
